magenta-lib/src/test/scala/magenta/input/DeploymentTypeResolverTest.scala

```diff
@@ -0,0 +1,55 @@
+package magenta.input
+
+import magenta.fixtures._
+import org.scalatest.{EitherValues, FlatSpec, Matchers}
+
+class DeploymentTypeResolverTest extends FlatSpec with Matchers with EitherValues {
+  val deployment = Deployment("bob", "stub-package-type", List("stack"), List("eu-west-1"), actions=None, "bob", "bob", Nil, Map.empty)
+  val deploymentTypes = List(stubPackageType(Seq("upload", "deploy")))
+
+  "validateDeploymentType" should "fail on invalid deployment type" in {
+    val deploymentWithInvalidType = deployment.copy(`type` = "invalidType")
+    val configError = DeploymentTypeResolver.validateDeploymentType(deploymentWithInvalidType, deploymentTypes).left.value
+    configError.context shouldBe "bob"
+    configError.message should include(s"Unknown type invalidType")
```

Comment:
If you're checking the exact message elsewhere, maybe you could do that here as well?